src/blob.rs

```diff
@@ -19,68 +28,154 @@ struct Header {
     flags: u64,
 }
 
-impl<K> Blob<K>
-where
-    K: Default,
-{
-    /// # Description
-    /// Create new blob from file
-    // @TODO more useful result
-    pub fn from_file<P: AsRef<Path>>(_path: P) -> Result<Self, ()> {
-        // @TODO implement
-        Ok(Self::default())
+#[derive(Debug)]
+pub struct WriteFuture {
+    file: fs::File,
+    buf: Option<Vec<u8>>,
+    current_offset: u64,
+}
+
+impl Future for WriteFuture {
+    type Output = Result<()>;
+
+    fn poll(mut self: Pin<&mut Self>, _waker: &Waker) -> Poll<Self::Output> {
+        let buf = self.buf.take().unwrap();
+        self.file.write_at(&buf, self.current_offset).unwrap();
+        Poll::Ready(Ok(()))
     }
+}
 
-    /// # Description
-    /// Writes given slice to file
-    // @TODO more useful result
-    pub fn write(&mut self, _record: Record<K>) -> Result<(), ()> {
-        // @TODO implement
-        Ok(())
+#[derive(Debug)]
+pub struct ReadFuture {
+    f: fs::File,
+    k: Vec<u8>,
+    offset: u64,
+    size: u64,
+}
+
+impl Future for ReadFuture {
+    type Output = Result<Record>;
+
+    fn poll(self: Pin<&mut Self>, _waker: &Waker) -> Poll<Self::Output> {
+        let mut buf = vec![0u8; self.size as usize];
+        self.f.read_at(&mut buf, self.offset).unwrap();
+        if let Some(rec) = Record::from_raw(&buf).take() {
+            Poll::Ready(Ok(rec))
+        } else {
+            Poll::Ready(Err(Error::RecordFromRawFailed))
+        }
     }
+}
 
+impl Blob {
     /// # Description
-    /// Reads record data, yields `Ok(Vec<u8>)` if read successful,
-    /// otherwise - `Err`
-    // @TODO more useful result
-    pub fn read(&self, _key: &K) -> Result<Record<K>, ()> {
-        // @TODO implement
-        Ok(Record::new())
+    /// Creates new blob file
+    pub fn open_new(path: PathBuf) -> Result<Self> {
+        Ok(Self {
+            header: Default::default(),
+            file: Some(
+                fs::OpenOptions::new()
+                    .create_new(true)
+                    .append(true)
+                    .read(true)
+                    .open(&path)
+                    .map_err(Error::OpenNew)?,
```

Comment:
We need to know what was the root of error - pass string with description

Reply:
[```Error::OpenNew```](https://github.com/qoollo/pearl/blob/feature/blob_init/src/blob.rs#L177) is a closure that wraps [```io::Result<T>```](https://doc.rust-lang.org/std/io/type.Result.html)